resources/idlharness.js

```diff
@@ -3298,15 +3302,23 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                test(() => {
+                    const validations = WebIDL2.validate(astArray);
+                    if (validations.length) {
+                        const message = validations.map(v => v.message).join("\n\n");
+                        throw new Error(message);
+                    }
+                }, "IDL validation");
```

Comment:
```suggestion
                }, "idl_test validation");
```

For consistency with "idl_test setup", but if you want to rename that too that'd be fine. These are subtests that could pass even if you don't implement the feature at all, so having test names that are easy to filter out helps.